### ISSUE TITLE
[BUG] Cannot generate Typescript fetch index.ts supporting files independently

### ISSUE THREAD

**Issue by alwaysbemark** [OpenAPITools/openapi-generator#10302]
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ ] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

For the Typescript Fetch generator, I need to be able to generate index.ts files inside the `models` directory without having to generate apis at all. Currently this is not possible because of what seems to be an oversight or a bug.

##### openapi-generator version

5.2.1

##### OpenAPI declaration file content or url

Irrelevant.

##### Generation Details

```
openApiGenerate {
    generatorName = "typescript-fetch"
    inputSpec = "foo/bar.yml"
    outputDir = "$projectDir/src"
    configOptions = [
            supportsES6: "true",
            typescriptThreePlus: "true"
    ]
    globalProperties = [
            apis: "",
            models: "",
            supportingFiles: "index.ts"
    ]
}
```

##### Steps to reproduce

Use the above configuration in Gradle to generate Typescript fetch code. This works beautifully but generates both the `apis` and `models` directory. I don't want the latter, so I delete `apis: ""` from the configuration above and the nested index.ts file inside `models` fails to generate properly.

##### Related issues/PRs

Haven't seen any.

##### Suggest a fix

Pretty sure I see what the problem is:

The index.ts file I'm after is being generated by method [TypeScriptFetchClientCodegen::postProcessOperationsWithModels](https://github.com/OpenAPITools/openapi-generator/blob/5d68bd6a03f0c48e838b4fe3b98b7e30858c0373/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptFetchClientCodegen.java#L572).
This method is only invoked by the parent class DefaultGenerator [here](https://github.com/OpenAPITools/openapi-generator/blob/5d68bd6a03f0c48e838b4fe3b98b7e30858c0373/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L1180). Unfortunately this method is skipped if we indicate that we don't want to generate apis [here](https://github.com/OpenAPITools/openapi-generator/blob/5d68bd6a03f0c48e838b4fe3b98b7e30858c0373/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L559). This seems to create this coupling between the generation of the apis and models index.ts supporting file.

**Comment by alwaysbemark** [comment]
Added sponsorship section cc @wing328 . Thanks so much for OpenAPI codegen!

**Comment by wing328** [comment]
What about the other way around by using .openapi-generator-ignore to skip files you don't want to generate?

Ref: https://github.com/OpenAPITools/openapi-generator/blob/master/docs/customization.md#ignore-file-format

**Comment by alwaysbemark** [comment]
Yup that'll work although it provides a less elegant solution for us in some ways. I'm leaving this open in case this is needed by anyone since it looks like that dependency should be broken. Thanks a lot!

**Comment by wing328** [comment]
openapi-geneator-ignore was implemented by @jimschubert . You can support his work directly via https://www.patreon.com/jimschubert

**Comment by alwaysbemark** [comment]
Thank you! Will look into sponsoring even if this doesn't get resolved. Happy to close this if you think that it isn't worth it!

**Comment by kereis** [comment]
Hi everyone, this issue seems to be still present. I tried to suppress generation via `.openapi-generator.ignore`, but the API files are still generated.

I'm using the maven plugin (version 6.0.1), placed a `.openapi-generator-ignore` in my `output` folder and set `<ignoreFileOverride />` in my `pom.xml`.

The `.openapi-generator-ignore` contains
```
/**/apis # `./apis` is in the root folder of `output`.
```

Did I do something wrong? I tried to circumvent it by simply removing the folder on `prepare-package`, but the `index.ts` needs to be updated as `./apis` is removed. On Windows, it's a bit tedious due to access locks on `index.ts` when working with an IDE.
